### PR TITLE
Added more advice for various methods UserMapper functions

### PR DESCRIPTION
**Pull request by arekmano**
- Renamed Aspect as it will be generic and wrapping all mappers.
- Added advice for makeNew and delete functions.
- Removed UserIdMap from UserMapper.
- This is part of the ongoing work on #5

### CODE REVIEW

File: app/mapper/UnitOfWork.py [hunk 1]
```diff
@@ -30,41 +30,41 @@ def registerDeleted(object):
...
             RoomMapper.erase(object)
-        if type(object) is Reservation:
+        if isinstance(object, Reservation):
             ReservationMapper.erase(object.getId())
-        if type(object) is Waiting:
+        if isinstance(object, Waiting):
             WaitingMapper.erase(object)
-        if type(object) is Timeslot:
+        if isinstance(object, Timeslot):
             TimeslotMapper.erase(object.getId())
```
Comment (lancelafontaine) [inline app/mapper/UnitOfWork.py]: Strange how the Mapper interface is not consistent here (some take object, other take ID).

File: app/mapper/UserMapper.py [hunk 2]
```diff
@@ -9,9 +8,8 @@ def __init__():
     pass
 
 
-def makeNew(name, password):
-    user = User(name, password)
```
Comment (lancelafontaine) [inline app/mapper/UserMapper.py]: Gotta wonder how this was even working previously when they were passing the wrong number of params.

Comment (arekmano) [inline app/mapper/UserMapper.py]: it for sure wasn't. They just wrote it to pass minimal code review from the prof

**Comment by lancelafontaine** [comment]
Awesome work. Only comment would be that it may make sense to include `Aspect.py` in a different outside directory, so that it is apparent that the original source is not with the aspect.

File: app/mapper/UnitOfWork.py (hunk 1 above)
Comment (arekmano) [inline app/mapper/UnitOfWork.py]: All actually take ID. The ones that aren't taking ID actually crash when executed.

Comment (lancelafontaine) [inline app/mapper/UnitOfWork.py]: 😱

**Comment by arekmano** [comment]
Only reason I did so is because it will be heavily tied with the mappers

**Pull request merged**